extensions-contrib/opentsdb-emitter/src/main/java/io/druid/emitter/opentsdb/OpentsdbSender.java

```diff
@@ -109,13 +141,11 @@ private void sendEvents()
     @Override
     public void run()
     {
-      while (running) {
-        if (!eventQueue.isEmpty()) {
-          OpentsdbEvent event = eventQueue.poll();
-          events.add(event);
-          if (events.size() >= flushThreshold) {
-            sendEvents();
-          }
+      while (!eventQueue.isEmpty() && !scheduler.isShutdown()) {
+        OpentsdbEvent event = eventQueue.poll();
+        events.add(event);
+        if (events.size() >= flushThreshold) {
+          sendEvents();
```

Comment:
If emitter has large amounts data continuously, eventQueue always has element, this will be multiple EventConsumer tasks at the same time. The object events is a List<OpentsdbEvent>, it's not safe when multiple tasks use it.

Reply:
Actually, I used `scheduleWithFixedDelay` not `scheduleAtFixedRate` in my code. There may be two EventConsumer tasks at the same time only in the case of `flush` method being called. And I fixed it.